src/common/components/repositories-home/index.js

```diff
@@ -59,11 +49,16 @@ class RepositoriesHome extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    const snapsLength = nextProps.snaps.ids.length;
-
-    if ((this.props.snaps.success !== nextProps.snaps.success)
-        || (snapsLength === 0)) {
-      this.fetchData(nextProps);
+    const { hasSnaps, snaps } = nextProps;
+    // if snaps stopped fetching
+    if ((this.props.snaps.isFetching !== snaps.isFetching) && !snaps.isFetching) {
```

Comment:
Rewritten the condition not to rely on `success` flag anymore, and always reload builds when snaps stopped fetching.